src/Field.tsx

```diff
@@ -128,20 +153,27 @@ export function Field({
     return children(bag);
   }
 
-  if (typeof component === 'string') {
+  if (component) {
+    if (typeof component === 'string') {
+      const { innerRef, ...rest } = props;
+      return React.createElement(
+        component,
+        { ref: innerRef, ...field, ...rest },
+        children
+      );
+    }
+    return React.createElement(component, { ...bag, ...props }, children);
+  }
+
+  if (typeof is === 'string') {
```

Comment:
Isn't this defaulted to `string` above on line 94?

```
as: is = 'input', // `as` is reserved in typescript lol
```

Reply:
Yep this is desired because this will return `true` for `as="select"` but `false` for `as={StyleComponent}`